Chessington.GameEngine/Pieces/Pawn.cs

```diff
@@ -5,12 +5,34 @@ namespace Chessington.GameEngine.Pieces
 {
     public class Pawn : Piece
     {
-        public Pawn(Player player) 
-            : base(player) { }
+        public Pawn(Player player)
+            : base(player)
+        {
+            
+        }
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            if ((board.FindPiece(this).Row == 1 && Player == Player.Black) || (board.FindPiece(this).Row == GameSettings.BoardSize - 1 && Player == Player.White))
```

Comment:
ok, this is a good one.
You are adding 1 move in both cases, so maybe you can move it out of this if check. Add it into a list. Check if you should add 2nd move into a list and then return whole list?